htlcswitch/switch.go

```diff
@@ -1075,6 +1079,23 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return s.failAddPacket(packet, linkErr.WireMessage(), addErr)
 		}
 
+		// We have found an outgoing link with a forwarding policy that
+		// the htlc satisfies. Before we proceed with forwarding, check
+		// whether the htlc will produce a circular forward through the
+		// same channel, and fail it if the switch does not allow
+		// forwards of this nature.
+		if linkErr := s.CheckCircularForward(
```

Comment:
A separate statement to call `CheckCircularForward` rather than embedding it in the if condition is probably more readable.